src/transformers/generation_utils.py

```diff
@@ -1013,12 +1015,13 @@ def generate(
         model_kwargs["output_hidden_states"] = output_hidden_states
         model_kwargs["use_cache"] = use_cache
 
-        if model_kwargs.get("attention_mask", None) is None:
+        has_attention_mask = "attention_mask" in set(inspect.signature(self.forward).parameters.keys())
```

Comment:
don't see a better way at the moment of verfying if model has `attention_mask` or not. But don't think calling `inspect` here once is a problem no?

Reply:
Don't see a problem with that.